src/vaults/VaultRouter.sol

```diff
@@ -123,7 +123,7 @@ contract VaultRouter is Auth, Multicall, Recoverable, IVaultRouter {
         require(!vault.supportsInterface(type(IERC7540Deposit).interfaceId), NonSyncDepositVault());
 
         VaultDetails memory vaultDetails = spoke.vaultDetails(vault);
-        SafeTransferLib.safeTransferFrom(vaultDetails.asset, owner, address(this), assets);
+        if (owner != address(this)) SafeTransferLib.safeTransferFrom(vaultDetails.asset, owner, address(this), assets);
```

Comment:
In which case the owner can be the same `VaultRouter`?

Reply:
Pseudocode: `multicall([claimRedeem(vaultA, receiver=VaultRouter), deposit(vaultB, owner=VaultRouter)]`